src/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs

```diff
@@ -185,12 +185,11 @@ public static void InvalidConstructors()
         }
 
         [Fact]
-        [ActiveIssue(1904, PlatformID.AnyUnix)]
         public static async Task StrangeFiles4()
         {
             ZipTest.IsZipSameAsDir(await StreamHelpers.CreateTempCopyStream(
                  ZipTest.strange("dataDescriptor.zip")), ZipTest.zfolder("normalWithoutBinary"), ZipArchiveMode.Update, true, false);
-	    }
+        }
```

Comment:
We have some cross-compat zip tests but they do not check the validity of the header tags, only the files/folders within the zip. I have an issue (https://github.com/dotnet/corefx/issues/4992) up for creating more cross-compat tests that cover scenarios that are valid on one system but invalid on the other. 

As part of that addition I'd like to increase the robustness of our ZipArchive validation (e.g. check the header tags for the central directory header) on the Zips that we're writing. It'll be a bit more in depth since we don't expose any API to get the OS version tag, so I'd planned on putting it in its own PR.